Tasks/google-play-release/GooglePlay.js

```diff
@@ -227,29 +239,400 @@ function updateTrack(packageName, track, versionCode, userFraction) {
+        var nextDir = path.join(metadataRootDirectory, nextLanguageCode);
+        addingAllMetadataPromise = addingAllMetadataPromise.then(function (languageCode, directory) {
+            tl.debug(`Processing metadata for language code ${languageCode}`);
+            return uploadMetadataWithLanguageCode.bind(this, languageCode, directory)();
+        }.bind(this, nextLanguageCode, nextDir));
+    }
+
+    return addingAllMetadataPromise;
+}
+
+/**
+ * Updates the details for a language with new information
+ * Assumes authorized
+ * @param {string} languageCode Language code (a BCP-47 language tag) of the localized listing to update
+ * @param {string} directory Directory where updated listing details can be found.
+ * @returns {Promise} A Promise that will return after all metadata updating operations are completed.
+ */
+function uploadMetadataWithLanguageCode(languageCode, directory) {
+    console.log(`Attempting to upload metadata in ${directory} for language code ${languageCode}`);
+
+    var updatingMetadataPromise;
+
+    var patchListingRequestParameters = {
+        language: languageCode
+    };
+
+    patchListingRequestParameters.resource = createPatchListingResource(languageCode, directory);
+    updatingMetadataPromise = edits.listings.patchAsync(patchListingRequestParameters);
+    
+    updatingMetadataPromise = updatingMetadataPromise.then(function () {
+        return addAllChangelogs.bind(this, languageCode, directory)();
+    });
+
+    updatingMetadataPromise = updatingMetadataPromise.then(function () {
+        return attachImages.bind(this, languageCode, directory)();
+    });
+
+    return updatingMetadataPromise;
+}
+
+/**
+ * Helper method for creating the resource for the edits.listings.patch method.
+ * @param {string} languageCode Language code (a BCP-47 language tag) of the localized listing to update
+ * @param {string} directory Directory where updated listing details can be found.
+ * @returns {Object} resource A crafted resource for the edits.listings.patch method.
+ *                              { languageCode: string, fullDescription: string, shortDescription: string, title: string, video: string }
+ */
+function createPatchListingResource(languageCode, directory) {
+    tl.debug(`Constructing resource to patch listing with language code ${languageCode} from ${directory}`);
+    var resourceParts = {
+        "fullDescription": "full_description.txt",
+        "shortDescription": "short_description.txt",
+        "title": "title.txt",
+        "video": "video.txt"
+    };
+
+    var resource = {
+        language: languageCode
+    };
+
+    for (var i in resourceParts) {
+        var file = path.join(directory, resourceParts[i]);
+        var fileContents;
+        try {
+            fileContents = fs.readFileSync(file);
+        } catch (e) {
+            tl.debug(`Failed to read metadata file ${file}. Ignoring...`);
+        }
+
+        resource[i] = fileContents.toString();
```

Comment:
They way this is written here, you are absolutely correct. The message is also wrong as it technically isn't ignoring the file, it's assuming it is empty, which I think is also incorrect.

I have moved this call into the try/catch block so that we actually skip adding if the file read fails.